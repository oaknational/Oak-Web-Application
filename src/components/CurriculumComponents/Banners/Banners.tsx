import HomePageBanner from "@/components/CurriculumComponents/HomePageBanner";

export default function Banners() {
  return (
    <HomePageBanner
      background="lemon"
      newText="Subjects added"
      ctaText="See curriculum plans"
      page="curriculum-landing-page"
    />
  );
}
