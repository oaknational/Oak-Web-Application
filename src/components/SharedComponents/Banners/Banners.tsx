import { OakFlex, OakP, OakPromoTag } from "@oaknational/oak-components";
import { useFeatureFlagEnabled } from "posthog-js/react";

import PromoBanner from "@/components/SharedComponents/PromoBanner";

export default function Banners() {
  const isMythbustingCampaignActive = useFeatureFlagEnabled(
    "mythbusting-campaign",
  );

  const muthbustingContent = (
    <OakP>Myths about teaching can hold you back</OakP>
  );
  const curriculumContent = (
    <OakFlex $gap="space-between-ssx">
      <OakPromoTag /> Subjects added
    </OakFlex>
  );

  const message = isMythbustingCampaignActive
    ? muthbustingContent
    : curriculumContent;
  const background = isMythbustingCampaignActive ? "lemon30" : "lemon";
  const ctaText = isMythbustingCampaignActive
    ? "Learn why"
    : "See curriculum plans";

  // TODO: add campaign page
  const page = isMythbustingCampaignActive
    ? "curriculum-landing-page"
    : "curriculum-landing-page";

  return (
    <PromoBanner
      background={background}
      ctaText={ctaText}
      page={page}
      message={message}
    />
  );
}
