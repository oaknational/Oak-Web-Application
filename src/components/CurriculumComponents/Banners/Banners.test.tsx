import Banners from "./Banners";

import render from "@/__tests__/__helpers__/renderWithTheme";

const isCycleTwoEnabled = jest.fn(() => false);
jest.mock("@/utils/curriculum/features", () => ({
  __esModule: true,
  isCycleTwoEnabled: (...args: []) => isCycleTwoEnabled(...args),
  default: {},
}));

describe("Banners", () => {
  test("when cycle 2 not enabled", async () => {
    isCycleTwoEnabled.mockReturnValue(false);
    const { container } = render(<Banners />);
    expect(container.innerHTML).toBe("");
  });
  test("when cycle 2 enabled", async () => {
    isCycleTwoEnabled.mockReturnValue(true);
    const { container } = render(<Banners />);
    expect(container).toHaveTextContent("See curriculum plans");
    expect(container).toMatchSnapshot();
  });
});
