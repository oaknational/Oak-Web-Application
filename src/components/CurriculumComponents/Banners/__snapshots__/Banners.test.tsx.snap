// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banners when cycle 2 enabled 1`] = `
<div>
  <div
    class="sc-ad0ad6f5-0 bZaRlP"
    role="banner"
  >
    <div
      class="sc-ad0ad6f5-0 sc-213e351e-0 dKiBog cQsapu"
    >
      <div
        class="sc-ad0ad6f5-0 sc-213e351e-0 dLtZMN glzuTZ"
      >
        <div
          class="sc-ad0ad6f5-0 sc-213e351e-0 IJRuG bvfibs"
        >
          <div
            aria-hidden="true"
            class="sc-ad0ad6f5-0 sc-213e351e-0 kukHQW cUwoYe"
            data-testid="new-icon"
          >
            <svg
              aria-hidden="true"
              class="sc-86d1d46f-0 kiKfWP"
              filter="drop-shadow(2px 2px 0px rgb(255 255 255 / 0.6))"
              height="100%"
              name="tag-promotional"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="/images/sprite/sprite.svg#tag-promotional"
              />
            </svg>
            <div
              class="sc-ad0ad6f5-0 sc-213e351e-0 SIsrt cUwoYe"
            >
              <span
                class="sc-kpDqfm dpkXxG"
              >
                New
              </span>
            </div>
          </div>
          <div
            class="sc-fqkvVR sc-eldPxv fmUONJ camYZD"
          >
            <span
              class="sc-50f33a24-0 ehBjeG"
            >
              NewÂ 
            </span>
             
            Subjects added
          </div>
        </div>
        <a
          class="sc-20316575-0 ldDVCs"
          href="/teachers/curriculum"
          role="link"
          title="See curriculum plans"
        >
          <div
            class="sc-ad0ad6f5-0 sc-213e351e-0 rqUDT BtlXC"
          >
            <span
              aria-hidden="true"
              class="sc-bea19ba6-0 bNTveO"
              data-testid="button-icon"
            >
              <svg
                aria-hidden="true"
                class="sc-86d1d46f-0 jOBmUC sc-bea19ba6-2 gvZQHW"
                height="100%"
                name="icon-background"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="/images/sprite/sprite.svg#icon-background"
                />
              </svg>
              <span
                class="sc-bea19ba6-1 fqRwJG"
              >
                <svg
                  aria-hidden="true"
                  class="sc-86d1d46f-0 eJtvEZ"
                  height="100%"
                  name="chevron-right"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="/images/sprite/sprite.svg#chevron-right"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="sc-ad0ad6f5-0 jIxoAR"
          >
            <div
              class="sc-ad0ad6f5-0 bZaRlP"
            >
              <span
                class="sc-ef29f35c-0 sc-859435d3-0 bIpPnV leLdFP"
              >
                See curriculum plans
              </span>
            </div>
          </div>
          <svg
            aria-hidden="true"
            class="sc-86d1d46f-0 jOBmUC sc-8aa75bfd-0 ewNWJE"
            height="100%"
            name="underline-1"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="/images/sprite/sprite.svg#underline-1"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="sc-86d1d46f-0 jOBmUC sc-8aa75bfd-2 jnsKyy"
            height="100%"
            name="underline-1"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="/images/sprite/sprite.svg#underline-1"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="sc-ad0ad6f5-0 depSXY"
      data-testid="hr"
      role="separator"
    >
      <svg
        aria-hidden="true"
        class="sc-86d1d46f-0 jOBmUC sc-5633ef5f-0 iPOyVF"
        height="100%"
        name="horizontal-rule"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="/images/sprite/sprite.svg#horizontal-rule"
        />
      </svg>
    </div>
  </div>
</div>
`;
